rust/src/web/graphql/blocks/mod.rs

```diff
@@ -32,49 +22,56 @@ impl BlocksQueryRoot {
     ) -> Result<Option<BlockWithCanonicity>> {
         let db = db(ctx);
 
+        // no query filters => get the best block
+        if query.is_none() {
+            return Ok(db.get_best_block().map(|b| {
+                b.map(|pcb| BlockWithCanonicity {
+                    block: pcb.into(),
+                    canonical: true,
+                })
+            })?);
+        }
+
         // Use constant time access if we have state hash
         if let Some(state_hash) = query.clone().and_then(|input| input.state_hash) {
-            let state_hash = &BlockHash::from(state_hash);
-            let pcb = match db.get_block(state_hash)? {
+            if !block::is_valid_state_hash(&state_hash) {
+                return Ok(None);
+            }
```

Comment:
I don't think this check is necessary. It's extra work for the same result.

Reply:
You're right, but it's less work for an invalid state hash